pkg/apis/core/validation/validation.go

```diff
@@ -6292,13 +6292,20 @@ func validateConfigMapNodeConfigSource(source *core.ConfigMapNodeConfigSource, f
 	return allErrs
 }
 
-// Validate compute resource typename.
+// Validate that the resource typename is qualified.
 // Refer to docs/design/resources.md for more details.
-func validateResourceName(value core.ResourceName, fldPath *field.Path) field.ErrorList {
+func validateResourceNameIsQualified(value core.ResourceName, fldPath *field.Path) field.ErrorList {
```

Comment:
So you are suggesting that instead of rejecting the feature gated names during validation, they should silently get removed the same way as gated fields are? And validation then doesn't need to check the feature gate?

I'm not sure whether that is better. When I discussed this with @liggitt [in 2023](https://github.com/kubernetes/kubernetes/pull/120611#discussion_r1368720017) I believe he wanted to check during validation.